src/index.ts

```diff
@@ -1,157 +1,152 @@
 
-}
+    appNames.forEach(appName => {
+        const sharedComponentsArray = getAppDependencies(appName);
+        console.log('sharedComponentsArray: ', sharedComponentsArray);
 
-function findAppDependencies(options: any): string[] {
-    const appNames = options.apps !== '.' ? `${options.apps}` : '';
-    const appNameArray: string | any[] = options.apps !== '' && options.apps?.includes(',') ? appNames.split(',') : [options.apps];
-    // console.log(`options.apps ${options.apps}`)
-    const dependentAppNames: string[] = []
-    const excludedAppNames = options.exclude !== '' ? `${options.exclude}` : '';
-    const excludedApps: string[] = options.exclude !== '' && options.exclude?.includes(',') ? excludedAppNames.split(',') : [options.exclude];
-    // console.log('excludedApps ',excludedApps)
-    if (options.apps.length > 0) {
-
-        for (let i = 0; i < appNameArray?.length; i++) {
-            const sharedComponentsArray = getAppDependencies(appNameArray[i]);
-            // console.log('sharedComponentsArray ',sharedComponentsArray)
-            const filteredAppsArray = sharedComponentsArray?.filter(obj =>
-                !excludedApps.some(substring => obj.path.includes(substring))
-            );
-            // console.log(filteredAppsArray);
-            if (filteredAppsArray) {
-                for (let j = 0; j < filteredAppsArray?.length; j++) {
-                    dependentAppNames.push(filteredAppsArray[j].path)
-                }
-            }
-            if (filteredAppsArray) {
-                console.log(`Dependencies for ${appNameArray[i]}:`, filteredAppsArray);
+        const filteredAppsArray = sharedComponentsArray?.filter(obj =>
+            !excludedApps.some(substring => obj.path.includes(substring))
+        );
+        console.log('filteredAppsArray: ', filteredAppsArray);
+
+        if (filteredAppsArray) {
+            for (let j = 0; j < filteredAppsArray?.length; j++) {
+                dependentAppNames.push(filteredAppsArray[j].path);
             }
         }
+    });
 
-    }
     return dependentAppNames;
 }
 
-function initAndSetSparseCheckoutForApp(options: any) {
-    const appFolderNames: string[] = findAppDependencies(options);
-    console.log('appFolderNames', appFolderNames)
-    const gitSparseCheckoutInitSuccess = executeCommand('git sparse-checkout init --cone');
-    console.log('gitSparseCheckoutInitSuccess ', gitSparseCheckoutInitSuccess)
-    if (!gitSparseCheckoutInitSuccess) {
+async function checkout(options: any) {
+    const isCommandAvailable = checkCommandAvailability('nx');
+    console.log('Called with options: ', options);
+
+    if (!options.apps) {
+        console.error('No apps specified for checkout.');
         return;
+    }
+
+    if (isCommandAvailable) {
+        const nxVersion = getNxVersion();
+        const command = 'nx';
+        let args: string[];
+
+        if (nxVersion >= 19) {
+            const base = getGitBaseCommit();
```

Comment:
This is not needed, when we do initial clone we will not have this. Just "nx affected:graph --file=output.json" will do